TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/ReplayEngine.java

```diff
@@ -129,6 +131,11 @@ public boolean isWorkOutstanding() {
         var start = timeShifter.transformSourceTimeToRealTime(originalStart);
         var end = timeShifter.transformSourceTimeToRealTime(originalEnd);
         var interval = numPackets > 1 ? Duration.between(start, end).dividedBy(numPackets-1) : Duration.ZERO;
+        metricsLogger.atSuccess()
+                .addKeyValue("requestId", requestKey.toString())
```

Comment:
similar comment to above - I would break this into two dimensions, connectionId and requestId.

Reply:
I like the idea of having both. I think that I would actually consider `connectionId` a dimension, and `requestId` more like a detail for an individual datapoint -- I can't imagine that we would ever use it as a dimension in cloudwatch, but it seems like a useful detail to include if a user needs to look into a specific event.

Reply:
s/dimension/datapoint/. How you slice and dice data is up to your metrics processing system.
Knowing that a request was from a new connection vs a reused one might have a big impact on some performance numbers, at least for some systems.  Anyway - whether you make something into dimensions or just datapoints is not an implementation detail that this class (or anything in the Replayer) needs to be concerned with.